src/main/java/org/llorllale/cactoos/matchers/Throws.java

```diff
@@ -66,35 +82,76 @@
      * @param msg The expected exception message.
      * @param type The expected exception type.
      */
-    public Throws(final String msg, final Class<? extends Exception> type) {
+    public Throws(
+        final String msg, final Class<? extends Exception> type
+    ) {
+        this(
+            type,
+            actual -> actual.equals(msg),
+            new UncheckedText(
+                new FormattedText("message '%s'", msg)
+            ).asString()
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param type The expected exception type.
+     * @param expectation The expectation which the exception message must
+     *  match.
+     */
+    public Throws(
+        final Class<? extends Exception> type,
+        final Func<String, Boolean> expectation
+    ) {
+        this(type, expectation, "expectation for message.");
+    }
+
+    /**
+     * Ctor.
+     * @param type The expected exception type.
+     * @param messageExpectation The expectation which the exception message
+     *  must match.
+     * @param expectationDescription The description of the message
+     *  expectation.
+     */
+    public Throws(
+        final Class<? extends Exception> type,
+        final Func<String, Boolean> messageExpectation,
+        final String expectationDescription
+    ) {
         super();
-        this.msg = msg;
         this.type = type;
+        this.messageExpectation = new UncheckedFunc<>(messageExpectation);
+        this.expectationDescription = expectationDescription;
     }
 
     @Override
     public void describeTo(final Description dsc) {
-        describe(dsc, this.type, this.msg);
+        describe(dsc, this.type, this.expectationDescription);
     }
 
     @Override
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    protected boolean matchesSafely(final Scalar<T> obj,
-        final Description dsc) {
+    protected boolean matchesSafely(
+        final Scalar<T> obj, final Description dsc
+    ) {
         // @checkstyle IllegalCatchCheck (20 lines)
         boolean matches;
         try {
             obj.value();
             matches = false;
             dsc.appendText("The exception wasn't thrown.");
         } catch (final Exception cause) {
-            describe(dsc, cause.getClass(), cause.getMessage());
-            if (this.type.isAssignableFrom(cause.getClass())
-                && this.msg.equals(cause.getMessage())) {
-                matches = true;
-            } else {
-                matches = false;
-            }
+            describe(
```

Comment:
@vzurauskas `Throws.describe` for clarity

Reply:
@ilyakharlamov Changed.